pkg/sql/executor_ir.go

```diff
@@ -79,28 +79,29 @@ func RunSQLProgram(sqlProgram string, modelDir string, session *pb.Session) *pip
 // *pipe.Reader, and remove the calls to log.Printf.
 func SubmitWorkflow(sqlProgram string, modelDir string, session *pb.Session) *pipe.Reader {
 	rd, wr := pipe.Pipe()
+	startTime := time.Now().Second()
 	go func() {
 		defer wr.Close()
-		err := submitWorkflow(wr, sqlProgram, modelDir, session)
-		if err != nil {
-			if err != pipe.ErrClosedPipe {
-				if err := wr.Write(err); err != nil {
-					log.Printf("submit workflow error(piping): %v", err)
-				}
+		wfID, err := submitWorkflow(wr, sqlProgram, modelDir, session)
+		defer log.Printf("Submit SQL program: %s\nuserID: %s\nworkflowID: %s\nspent: %d\nerror:%v", sqlProgram, session.UserId, wfID, time.Now().Second()-startTime, err)
```

Comment:
Why use `defer` here, the "submit" action should be logged right after the action?

Reply:
Using `defer` to calculate the duration time.